tests/test_api.py

```diff
@@ -308,6 +310,104 @@ def it_contains_only_the_sparse_paths(config):
             expect(dir_listing).contains('src')
             expect(len(dir_listing) == 1)
 
+    def describe_default_group():
+      @pytest.fixture
+      def config_with_default_group(config):
+        config.datafile.text = strip(
+        """
+        location: deps
+        sources:
+          - name: gitman_1
+            type: git
+            repo: https://github.com/jacebrowning/gitman-demo
+            sparse_paths:
+              -
+            rev: example-branch
+            link:
+            scripts:
+              -
+          - name: gitman_2
+            type: git
+            repo: https://github.com/jacebrowning/gitman-demo
+            sparse_paths:
+              -
+            rev: example-tag
+            link:
+            scripts:
+              -
+        groups:
+          - name: main
+            members:
+              - gitman_1
+          - name: secondary
+            members:
+              - gitman_2
+        default_group: 'main'
+        """
+        )
+        config.datafile.load()
+
+        return config
+
+      @pytest.fixture
+      def config_without_default_group(config):
+        config.datafile.text = strip(
+        """
+        location: deps
+        sources:
+          - name: gitman_1
+            type: git
+            repo: https://github.com/jacebrowning/gitman-demo
+            sparse_paths:
+              -
+            rev: example-branch
+            link:
+            scripts:
+              -
+          - name: gitman_2
+            type: git
+            repo: https://github.com/jacebrowning/gitman-demo
+            sparse_paths:
+              -
+            rev: example-tag
+            link:
+            scripts:
+              -
+        groups:
+          - name: main
+            members:
+              - gitman_1
+          - name: secondary
+            members:
+              - gitman_2
+        default_group: ''
+        """
+        )
+        config.datafile.load()
+
+        return config
+
+      def it_installs_default_group(config_with_default_group):
+        expect(gitman.install(depth=1, force=True)) == True
+        expect(os.path.exists(os.path.join(config_with_default_group.location, 'gitman_1'))) == True
+        expect(os.path.exists(os.path.join(config_with_default_group.location, 'gitman_2'))) == False
+
+      def it_installs_all_sources_when_no_default_group(config_without_default_group):
+        expect(gitman.install(depth=1, force=True)) == True
+        expect(os.path.exists(os.path.join(config_without_default_group.location, 'gitman_1'))) == True
+        expect(os.path.exists(os.path.join(config_without_default_group.location, 'gitman_2'))) == True
+
+      # def it_installs_default_group_for_children(config_with_default_group):
```

Comment:
I tested these cases manually by building gitman as an exe, but to test them as 'unit-tests' would require adding a new gitman-demo repo which pulls in child dependencies. This could be tested with mocks, but this test file seems to err on the side of not using mocks.